File: performance-comparison/benchmarks/numerical/python/monte_carlo_pi.py
import sys

def estimate_pi(iters):
    state = 6364136223846793005
    mask = (1 << 64) - 1
    inside = 0
    for _ in range(iters):
        state = (state * 6364136223846793005 + 1442695040888963407) & mask
        x = (state >> 11) / (1 << 53)
        state = (state * 6364136223846793005 + 1442695040888963407) & mask
        y = (state >> 11) / (1 << 53)
        if x * x + y * y <= 1.0:
            inside += 1
    return 4.0 * inside / iters

n = int(sys.argv[1]) if len(sys.argv) > 1 else 1000
print(f"{estimate_pi(n):.6f}")

File: performance-comparison/benchmarks/numerical/clojure/monte_carlo_pi.clj
(let [iters (Long/parseLong (first *command-line-args*))
      inside (loop [i 0 state (long 6364136223846793005) inside 0]
               (if (>= i (* 2 iters))
                 inside
                 (let [s1 (unchecked-add (unchecked-multiply state 6364136223846793005)
                                         1442695040888963407)
                       x  (/ (double (unsigned-bit-shift-right s1 11))
                              (double (bit-shift-left 1 53)))
                       s2 (unchecked-add (unchecked-multiply s1 6364136223846793005)
                                         1442695040888963407)
                       y  (/ (double (unsigned-bit-shift-right s2 11))
                              (double (bit-shift-left 1 53)))]
                   (recur (+ i 2) s2 (if (<= (+ (* x x) (* y y)) 1.0)
                                       (inc inside) inside)))))]
  (printf "%.6f%n" (* 4.0 inside (/ 1.0 iters))))
